database/src/main/postgres/runs/V1.8.3__get_partitioning_checkpoint.sql

```diff
@@ -0,0 +1,112 @@
+    IN i_checkpoint_name           TEXT DEFAULT NULL,
+    OUT status                     INTEGER,
+    OUT status_text                TEXT,
+    OUT id_checkpoint              UUID,
+    OUT checkpoint_name            TEXT,
+    OUT measure_name               TEXT,
+    OUT measure_columns            TEXT[],
+    OUT measurement_value          JSONB,
+    OUT checkpoint_start_time    TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_end_time        TIMESTAMP WITH TIME ZONE
+)
+    RETURNS SETOF record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.get_partitioning_checkpoints(JSONB, INT, TEXT)
+--      Returns all the checkpoint for the given partitioning and checkpoint name
+--
+-- Parameters:
+--      i_partitioning      - partitioning for requested checkpoints
+--      i_limit             - limit of the number of checkpoints to return
+--      i_checkpoint_name   - name of the checkpoint to filter by
+--
+-- Returns:
+--      status                  - Status code
+--      status_text             - Status message
+--      id_checkpoint           - ID of the checkpoint
+--      checkpoint_name         - Name of the checkpoint
+--      measure_name            - Name of the measure
+--      measure_columns         - Columns of the measure
+--      measurement_value       - Value of the measurement
+--      checkpoint_start_time   - Time of the checkpoint
+--      checkpoint_end_time     - End time of the checkpoint computation
+--
+-- Status codes:
+--      11 - OK
+--      16 - No record found for the given partitioning
+--      41 - Partitioning not found
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_partitioning                    BIGINT;
+BEGIN
+    _fk_partitioning = runs._get_id_partitioning(i_partitioning);
+
+    IF _fk_partitioning IS NULL THEN
+        status := 41;
+        status_text := 'Partitioning not found';
+        RETURN NEXT;
+        RETURN;
+    END IF;
+
+    RETURN QUERY
+        SELECT
+            11 AS status,
+            'Ok' AS status_text,
+            c.id_checkpoint,
+            c.checkpoint_name,
+            md.measure_name,
+            md.measured_columns,
+            m.measurement_value,
+            c.process_start_time AS checkpoint_start_time,
+            c.process_end_time AS checkpoint_end_time
+        FROM
+            runs.checkpoints c
+        JOIN
+            runs.measurements m ON c.id_checkpoint = m.fk_checkpoint
+        JOIN
+            runs.measure_definitions md ON m.fk_measure_definition = md.id_measure_definition
+        WHERE
+            c.fk_partitioning = _fk_partitioning
+        AND
+            (i_checkpoint_name IS NULL OR c.checkpoint_name = i_checkpoint_name)
+        ORDER BY
+            c.process_start_time,
+            c.id_checkpoint
+        LIMIT nullif(i_limit, 0);
+
+        IF NOT FOUND THEN
+            status := 16;
```

Comment:
okay this can be now deleted, somewhere we agreed that it would be more difficult to work with it on user's side

Reply:
You mean the limit part or line 98 going forward.

Reply:
Status 16 related code